src/extensions/Statiq.ApiClient/ReadApiWithClient.cs

```diff
@@ -0,0 +1,149 @@
+    /// will be sent for each input document.
+    /// </remarks>
+    /// <category>Metadata</category>
+    public class ReadApiWithClient<TClient> : ParallelSyncModule
+        where TClient : class
+    {
+        private readonly Dictionary<string, Func<IDocument, IExecutionContext, TClient, object>> _requests
+            = new Dictionary<string, Func<IDocument, IExecutionContext, TClient, object>>();
+
+        private readonly TClient _client;
+
+        private Action<IDocument, IExecutionContext, TClient> _init;
+
+        private string _clientName;
+
+        /// <summary>
+        /// Creates a connection to the API using client.
+        /// </summary>
+        /// <param name="client">The API client to use.</param>
+        public ReadApiWithClient(TClient client)
+        {
+            _client = client ?? throw new ArgumentException("Argument is null", nameof(client));
+        }
+
+        /// <summary>
+        /// Initialize the API client.
+        /// </summary>
+        /// <param name="init">An API client initialization action.</param>
+        /// <returns>The current module instance.</returns>
+        public ReadApiWithClient<TClient> WithClientInitialization(Action<TClient> init)
+        {
+            if (init == null)
+            {
+                throw new ArgumentException("Argument is null or empty", nameof(init));
+            }
+
+            _init = (doc, ctx, client) => init(client);
+            return this;
+        }
+
+        /// <summary>
+        /// Initialize the API client.
+        /// </summary>
+        /// <param name="init">An API client initialization action.</param>
+        /// <returns>The current module instance.</returns>
+        public ReadApiWithClient<TClient> WithClientInitialization(Action<IExecutionContext, TClient> init)
+        {
+            if (init == null)
+            {
+                throw new ArgumentException("Argument is null or empty", nameof(init));
+            }
+
+            _init = (doc, ctx, client) => init(ctx, client);
+            return this;
+        }
+
+        /// <summary>
+        /// Initialize the API client.
+        /// </summary>
+        /// <param name="init">An API client initialization action.</param>
+        /// <returns>The current module instance.</returns>
+        public ReadApiWithClient<TClient> WithClientInitialization(Action<IDocument, IExecutionContext, TClient> init)
+        {
+            if (init == null)
+            {
+                throw new ArgumentException("Argument is null or empty", nameof(init));
+            }
+
+            _init = init.ThrowIfNull(nameof(init));
+            return this;
+        }
+
+        /// <summary>
+        /// Submits the API client name (for logging).
+        /// </summary>
+        /// <param name="name">Client name.</param>
+        /// <returns>The current module instance.</returns>
+        public ReadApiWithClient<TClient> WithClientName(string name)
```

Comment:
Not sure how I feel about a fluent configuration method that only influences logging. I'm not strongly opposed to it, but I don't think that convention is used anywhere else. I wonder if passing a name would work better as optional constructor overloads.

Reply:
It is a better idea, i am agree.

Reply:
The new constructors are perfect - I like setting a default "API" value for the name but allowing that to be provided if needed.